coramin/utils/ordered_containers.py

```diff
@@ -0,0 +1,122 @@
+from collections import abc
+from collections import OrderedDict
```

Comment:
Is this necessary anymore since dicts are insertion ordered now? If we do not support 3.5, we could use ComponentMap and ComponentSet from Pyomo.